src/formatDate.js

```diff
@@ -8,7 +8,51 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let day = 0;
+  let month = 0;
+  let year = 0;
+
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  let result = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = i;
+    }
+    if (fromFormat[i] === 'MM') {
+      month = i;
+    }
+    if (fromFormat[i] === 'DD') {
+      day = i;
+    }
+  }
```

Comment:
The loop used to assign the index positions to 'day', 'month', and 'year' can be optimized. Since 'fromFormat' will always contain 'YYYY', 'MM', and 'DD' in some order, you can directly assign the index positions without looping through the array.